modules/gdscript/gdscript_analyzer.h

```diff
@@ -36,11 +36,32 @@
 #include "core/templates/set.h"
 #include "gdscript_cache.h"
 #include "gdscript_parser.h"
+#include <map>
 
 class GDScriptAnalyzer {
 	GDScriptParser *parser = nullptr;
 	HashMap<String, Ref<GDScriptParserRef>> depended_parsers;
 
+	struct RenamedIdentifier {
+		String old_name;
+		String new_name;
+		GDScriptParser::Node::Type type;
+	};
+
+#ifdef DEBUG_ENABLED
+	// <Base class, renamed identifiers>
+	const std::map<String, Vector<RenamedIdentifier>> renamed_identifiers = {
```

Comment:
Godot tries to avoid using std. In addition there should already be a list of 3.x -> 4.0 renames in the project upgrader. Maybe try to look into reusing that list.